apis/python/src/tiledbsoma/io/outgest.py

```diff
@@ -300,44 +311,73 @@ def to_anndata(
-            obsm[key] = _extract_obsm_or_varm(
-                measurement.obsm[key], "obsm", key, nobs, obsm_width_hints
+            obsm[key] = tp.submit(
+                _extract_obsm_or_varm,
+                measurement.obsm[key],
+                "obsm",
+                key,
+                nobs,
+                obsm_width_hints,
             )
 
     varm = {}
     if "varm" in measurement:
         varm_width_hints = obsm_varm_width_hints.get("obsm", {})
         for key in measurement.varm.keys():
-            varm[key] = _extract_obsm_or_varm(
-                measurement.varm[key], "varm", key, nvar, varm_width_hints
+            varm[key] = tp.submit(
+                _extract_obsm_or_varm,
+                measurement.varm[key],
+                "varm",
+                key,
+                nvar,
+                varm_width_hints,
             )
 
     obsp = {}
     if "obsp" in measurement:
+
+        def load_obsp(measurement: Measurement, key: str, nobs: int) -> sp.csr_matrix:
+            return conversions.csr_from_coo_table(
+                measurement.obsp[key].read().tables().concat(), nobs, nobs
+            )
+
         for key in measurement.obsp.keys():
-            matrix = measurement.obsp[key].read().tables().concat().to_pandas()
-            obsp[key] = conversions.csr_from_tiledb_df(matrix, nobs, nobs)
+            obsp[key] = tp.submit(load_obsp, measurement, key, nobs)
 
     varp = {}
     if "varp" in measurement:
+
+        def load_varp(measurement: Measurement, key: str, nvar: int) -> sp.csr_matrix:
+            return conversions.csr_from_coo_table(
+                measurement.varp[key].read().tables().concat(), nvar, nvar
+            )
+
         for key in measurement.varp.keys():
-            matrix = measurement.varp[key].read().tables().concat().to_pandas()
-            varp[key] = conversions.csr_from_tiledb_df(matrix, nvar, nvar)
+            varp[key] = tp.submit(load_varp, measurement, key, nvar)
 
-    uns: UnsMapping = {}
+    uns_future: Future[Dict[str, FutureUnsDictNode]] | None = None
     if "uns" in measurement:
         s = _util.get_start_stamp()
         uns_coll = cast(Collection[Any], measurement["uns"])
         logging.log_io(None, f"Start  writing uns for {uns_coll.uri}")
-        uns = _extract_uns(
-            uns_coll,
-            uns_keys=uns_keys,
-        )
+        uns_future = tp.submit(_extract_uns, uns_coll, uns_keys=uns_keys)
         logging.log_io(
             None,
             _util.format_elapsed(s, f"Finish writing uns for {uns_coll.uri}"),
         )
 
+    # Resolve all futures
+    obsm = _resolve_futures(obsm)
+    varm = _resolve_futures(varm)
+    obsp = _resolve_futures(obsp)
+    varp = _resolve_futures(varp)
+    anndata_X = anndata_X_future.result() if anndata_X_future is not None else None
+    uns: UnsDict = (
+        _resolve_futures(uns_future.result(), deep=True)
+        if uns_future is not None
+        else {}
+    )
```

Comment:
```suggestion
    uns: UnsDict = _resolve_futures(uns_future.result(), deep=True) or {}
```

Reply:
But it looks like `_resolve_futures` only returns a `Dict` and not `Optional[Dict]`, so could be further simplified to `uns: UnsDict = _resolve_futures(uns_future.result(), deep=True)`?